sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/jsonpatch/JsonPatchDocument.java

```diff
@@ -65,19 +87,17 @@ public JsonPatchDocument appendAdd(String path, String rawJsonValue) {
      *
      * <p><strong>Code Samples</strong></p>
      *
-     * {@codesnippet com.azure.core.experimental.jsonpatch.JsonPatchDocument.appendReplace#String-String}
+     * {@codesnippet com.azure.core.experimental.jsonpatch.JsonPatchDocument.appendReplace#String-Object}
      *
      * @param path The path to replace.
-     * @param rawJsonValue The raw JSON value to use as the replacement.
+     * @param value The value to use as the replacement.
      * @return The updated JsonPatchDocument object.
-     * @throws NullPointerException If {@code path} or {@code rawJsonValue} is null.
+     * @throws NullPointerException If {@code path} is null.
      */
-    public JsonPatchDocument appendReplace(String path, String rawJsonValue) {
+    public JsonPatchDocument appendReplace(String path, Object value) {
```

Comment:
I think switching the behavior of this parameter based on ctor parameter might be a little confusing.

Imagine you were replacing some trivial properties in the beginning but now wanting to pass a serialized object into another property. Passing the serializer into ctor would change the behavior of all existing code.

Reply:
Do we foresee it being a common scenario for multiple seriliazers to be used when interacting with a single `JsonPatchDocument`? With the pattern being the patch document is tied to a single serializer I would think that a different document would be used.

Another possible solution to this problem, given `JsonPatchOperation` is in public API is to add an `appendOperation(JsonPatchOperation operation)` API where the customer would set the `optionalValue` manually if they wanted to use multiple serializers for a single document.

Reply:
I was confused as to what default behavior is.